### PR TITLE
Plaintext components for adblock lists

### PR DESCRIPTION
**Pull request by antonok-edm**
Resolves https://github.com/brave/brave-browser/issues/25363

This PR makes the browser use the newly created plaintext components for adblock filter list data. It also uses new individual components for adblock resources and the regional catalog, so that these can be updated independently of the filter lists (which change much more often).

Commit-by-commit review is recommended.

### Performance analysis

Benchmarking `adblock-rust` on my own computer shows a minor perf impact related to list parsing at startup, of an additional ~56ms or 32% regression for the entire set of default lists. Note that an additional change can allow adblock-rust to cache compiled DATs in uncompressed format after updates, resulting in a significant improvement to engine load time over the current `master` branch. I've left the caching for a followup to minimize the number of changes in this PR.

The benchmark results below show the engine load times for the current `master` configuration, the configuration as of this PR, and the configuration with caching, respectively (middle time value is average).

```
from_dat                 time:   [173.66 ms 174.88 ms 176.32 ms]
from_plaintext           time:   [228.08 ms 230.97 ms 233.52 ms]
from_uncompressed_dat    time:   [46.191 ms 46.777 ms 47.546 ms]
```

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

The main change in this PR switches default and regional components to load filters as plaintext rather than from a DAT. This mechanism has already been well-tested in production thanks to the `FiltersProvider` architecture shared with the subscription lists and custom filters, as well as existing unit and browsertests.

### CODE REVIEW

File: ios/browser/api/brave_shields/adblock_filter_list_catalog_entry.mm [hunk 1]
```diff
@@ -18,6 +18,8 @@ @interface AdblockFilterListCatalogEntry ()
 @property(nonatomic, copy) NSString* componentId;
 @property(nonatomic, copy) NSString* base64PublicKey;
 @property(nonatomic, copy) NSString* desc;
+@property(nonatomic, copy) NSString* iosComponentId;
+@property(nonatomic, copy) NSString* iosBase64PublicKey;
```
[the remark this answers is not in the document]
Comment (antonok-edm) [reply ios/browser/api/brave_shields/adblock_filter_list_catalog_entry.mm:22]: iOS is still using the precompiled DAT components, but @cuba is working on [swapping them over as well](https://github.com/brave/brave-ios/issues/5975). In the meantime I figured it's clearer to expose a 1:1 view of the `FilterListCatalogEntry` struct, and we can remove the fields from the `brave-core` side as well.

I did actually try to overwrite the original `componentId` and `base64PublicKey` fields originally, but the major difficulty is that we still need to keep track of both sets of info on the `brave-core` side and then it's suddenly impossible to pass this struct across the FFI without losing information.

**Review by kylehickinson — APPROVED** [review]
iOS++

File: components/brave_shields/browser/ad_block_component_filters_provider.cc [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+#include "base/files/file_path.h"
+#include "base/logging.h"
+#include "base/task/thread_pool.h"
+#include "brave/components/brave_shields/browser/ad_block_component_installer.h"
+#include "brave/components/brave_shields/browser/filter_list_catalog_entry.h"
+#include "components/component_updater/component_updater_service.h"
+#include "content/public/browser/browser_task_traits.h"
+
+#define LIST_FILE "list.txt"
```
Comment (bridiver) [inline components/brave_shields/browser/ad_block_component_filters_provider.cc:20]: why a define for this and not a constexpr?

File: components/brave_shields/browser/filter_list_catalog_entry.cc [hunk 3]
```diff
@@ -54,15 +84,19 @@ FilterListCatalogEntry::FilterListCatalogEntry(
     const std::string& support_url,
     const std::string& component_id,
     const std::string& base64_public_key,
-    const std::string& desc)
+    const std::string& desc,
+    const std::string& ios_component_id,
```
Comment (bridiver) [inline components/brave_shields/browser/filter_list_catalog_entry.cc:88]: I don't understand why we have a separate entry for ios_component_id. Shouldn't we just initialize this with the correct component id for ios vs desktop?

Comment (antonok-edm) [reply components/brave_shields/browser/filter_list_catalog_entry.cc:88]: See https://github.com/brave/brave-core/pull/15077#discussion_r976700939 - this struct is just a 1:1 C++ representation of what we're shipping from the component server and we'll remove the `ios_` fields altogether once https://github.com/brave/brave-ios/issues/5975 is finished

File: components/brave_shields/browser/ad_block_component_filters_provider.cc (hunk 2 above)
Comment (antonok-edm) [reply components/brave_shields/browser/ad_block_component_filters_provider.cc:20]: that's how it was before, but good opportunity to update it

Comment (antonok-edm) [reply components/brave_shields/browser/ad_block_component_filters_provider.cc]: Updated this one and the `regional_catalog.json` one as well

**Review by bridiver — APPROVED** [review]

**Pull request merged**